### ISSUE TITLE
Bump upper shapely limit version to 1.7.0

### ISSUE THREAD

**Issue by richpsharp** [natcap/invest#200]
Right now it's 1.7.0 but that seems to work okay in another build I have so lets try to relax it.

**Comment by phargogh** [comment]
Could we leave this open?  I'd love to loosen this requirement, and the issues we have in the linked PR look to be tractable:

* https://github.com/natcap/invest/pull/201/checks?check_run_id=832208025 Looks like there are just 3 errors in the tests (haven't dug into it yet)
* https://github.com/natcap/invest/pull/201/checks?check_run_id=832208013 Looks like it's crashing hard, but in a way that makes me wonder if it's a GDAL/GEOS issue rather than a shapely/GEOS issue.
* https://github.com/natcap/invest/runs/832207633?check_suite_focus=true Looks like the GEOS DLL might have been renamed as well.  We'll need to handle that case, too.

**Comment by richpsharp** [comment]
Ha, sure. Can I assign it to you?

**Comment by dcdenu4** [comment]
It looks like the conda-forge feedstock currently has 1.7.1 as the latest version for shapely. I also don't see shapely 1.8.0 released on the shapely github repo: https://github.com/Toblerity/Shapely.

It certainly would be nice to move to 1.7.1 though.

**Comment by dcdenu4** [comment]
PR #314 moves to build invest with a conda environment workflow and when that gets merged it might be worth revisiting relaxing the constraint for shapely. Shapely 1.7.1 works fine locally on my fork and I can run / test the UCM model.

**Comment by dcdenu4** [comment]
I just played around with some conda environments and it looks like bumping to `shapely >= 1.7.0` pushes `geos = 3.8.1` ( as opposed to `3.8.0`). This causes some InVEST tests to fail like in `test_ucm.py` function `test_ucm_regression_factors`:
```python
E   AssertionError: 3520213.280928277 != 3520212.4242880843 within 4 places (0.8566401926800609 difference) : avd_eng_cn should be close to 3520212.424288, actual: 3520213.280928
```

I think the bump in `geos` is also what happens when bumping `gdal` to `3.1.2`, which causes tests to fail. So we'll have to refactor some tests for the new `geos` version.

**Comment by phargogh** [comment]
Just wanted to note that while reviewing #318 , using PyPI-provided Shapely binaries (`shapely==1.6.4.post2`) with `conda-forge`-provided `gdal==3.1.3` caused some segfaults in watershed delineation.  Installing the latest PyPI-provided shapely (currently 1.7.1) works great.

**Comment by phargogh** [comment]
@dcdenu4 wouldn't we still want the shapely upper-limit be 1.8.0?  It's currently defined as `Shapely>=1.6.4,<1.7.0`

**Comment by dcdenu4** [comment]
@phargogh I think the latest shapely is 1.7.1! So ideally, we'd want `Shapely>=1.7.0`. We'll just need to eye spot some test changes to make sure we're comfortable with the subtle under the hood changes with the updated vector geometry algorithms that `geos>=3.8.1` bring.

**Comment by dcdenu4** [comment]
Update: I'm not so sure `geos` is causing the tests to fail. I actually think my current conda env is forcing `proj 7` on me! That makes way more sense for why tests would be slightly different.

**Comment by phargogh** [comment]
Surprising that GEOS would be built against PROJ!  Ah well.  OK, so then assuming that these subtle changes in outputs are acceptable, then we'd be bumping this requirement to `shapely>=1.7.0,<1.8.0`?

**Comment by dcdenu4** [comment]
This was fixed in PR #350
